lib/file-source/src/file_watcher.rs

```diff
@@ -102,6 +83,14 @@ impl FileWatcher {
         self.findable
     }
 
+    pub fn get_file_position(&mut self) -> FilePosition {
```

Comment:
This feels like something we could maintain per-file rather than lookup every time. Especially if we're using it in a somewhat tight loop (for every line read from every file).

Reply:
You are right. Will do.